udata_front/theme/gouvfr/assets/js/pages/PublishingForm/Step3AddFiles.vue

```diff
@@ -0,0 +1,305 @@
+              class="fr-fieldset__element min-width-0"
+              :accordion="publishFileAccordionId"
+              @blur="touch"
+            >
+              <Container
+                color="alt-grey"
+                class="fr-grid-row fr-grid-row--middle flex-direction-column"
+                v-if="files.length === 0"
+              >
+                <h3 class="fr-text--md fr-text--bold fr-m-0 fr-mb-2w">{{ $t("Add your first files") }}</h3>
+                <UploadModalButton
+                  groupClass="fr-grid-row flex-direction-column fr-grid-row--middle"
+                  :label="$t('Add files')"
+                  :title="$t('Add files (opens a modal dialog)')"
+                  :multiple="true"
+                  :required="true"
+                  :hasError="stateHasError('files')"
+                  :errorText="getErrorText('files')"
+                  @change="addFiles"
+                />
+              </Container>
+              <template v-else>
+                  <FileCard
+                    v-for="(resource, index) in files"
+                    class="fr-mb-3v"
+                    :filename="resource?.file?.name || resource.url"
+                    :filesize="resource.filesize"
+                    :format="resource.format"
+                    :lastModified="resource?.file?.lastModified"
+                    :missingMetadata="!resource.title || !resource.description"
+                    :title="resource.title || resource.file?.name || ''"
+                    @delete="removeFile(index)"
+                    @edit="$emit('editFile', files, index)"
+                    :loading="isLoading(resource)"
+                    :hideActions="isLoaded(resource)"
+                  />
+                <div class="fr-grid-row fr-grid-row--center">
+                  <ButtonLoader :width="114" v-if="loading"/>
+                  <UploadModalButton
+                    :label="$t('Add files')"
+                    :title="$t('Add files (opens a modal dialog)')"
+                    :multiple="true"
+                    :required="true"
+                    :hasError="stateHasError('files')"
+                    :errorText="getErrorText('files')"
+                    @change="addFiles"
+                    v-else
+                  />
+                </div>
+              </template>
+            </LinkedToAccordion>
+          </fieldset>
+          <Alert type="error" v-if="errors.length" class="fr-mt-n2w fr-mb-2w">
+            <template #title>{{ $t("An error occured | Some errors occured", errors.length) }}</template>
+            <ul v-if="errors.length > 1">
+              <li v-for="error in errors">{{ error }}</li>
+            </ul>
+            <p v-else> {{ errors[0] }}</p>
+          </Alert>
+          <div class="fr-grid-row fr-grid-row--right">
+            <ButtonLoader :width="66" v-if="loading"/>
+            <button class="fr-btn" @click="submit" v-else>
+              {{ $t("Next") }}
```

Comment:
In one of my run through, I had an error during the "Next" from step 3 to 4. I deleted the file that could not be uploaded due to its format. I re-clicked on Next, but I had two issues :
* I had my files uploaded twice
* the URL behind "Save as draft" was erroneous due to the dataset `id` being concatenated twice in path

Reply:
Fixed : files already loaded are skip during upload, "save as draft" url uses the same base url on each upload